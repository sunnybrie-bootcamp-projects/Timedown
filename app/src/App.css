@import url("https://use.typekit.net/inp5cfh.css");

:root {
  /* COLORS */
  --white: ghostwhite;
  --yellow: rgba(232, 196, 104, 1);
  --navy: rgba(39, 71, 84, 1);
  --green: rgba(43, 158, 145, 1);
  --orange: rgba(245, 164, 98, 1);
  --bloodorange: rgba(232, 111, 81, 1);
  --softblue: rgba(144, 209, 202, 1);

  --transparent_yellow: rgba(232, 196, 104, 0.7);
  --transparent_navy: rgba(39, 71, 84, 0.7);
  --transparent_green: rgba(43, 158, 145, 0.5);
  --transparent_orange: rgba(245, 164, 98, 0.7);
  --transparent_bloodorange: rgba(232, 111, 81, 0.7);
  --transparent_softblue: rgba(144, 209, 202, 0.85);
  --transparent_white: rgba(248, 248, 255, 0.55);
  --invisible: rgba(0, 0, 0, 0);

  --shadow: rgba(34, 62, 73, 0.605);
  --small_shadow: 0.125em 0.125em 0.25em var(--shadow);
  --medium_shadow: 0.25rem 0.25rem 0.5rem var(--shadow);
  --large_shadow: 0.5rem 0.5rem 0.75rem var(--shadow);

  --thematic: quicksand, sans-serif;
  --regular: freight-neo-pro, sans-serif;

  --app_border: 3px solid var(--green);
  --notToggled_border: 3px solid var(--transparent_green);
  --small_radius: 0.5rem 0.5rem 0.5rem;
  --large_radius: 1rem 1rem 1rem;

  --paper: url("./assets/papertexture.jpeg");
}

html {
  height: 100vh;
  background: url("./assets/backgroundsplash.png") var(--softblue);
  background-size: contain;
  background-position-x: right;
  background-position-y: bottom;
  background-repeat: no-repeat;
  background-blend-mode: exclusion;

  overflow-x: hidden;
  overflow-y: auto;
}

body {
  font-family: freight-neo-pro, sans-serif;
  font-weight: 400;
  font-style: normal;
  color: var(--navy);
}

body::-webkit-scrollbar {
  width: 1em;

  background-color: rgb(140, 67, 49);
}

body::-webkit-scrollbar-track {
  box-shadow: inset 0 0 3px var(--shadow);
}

body::-webkit-scrollbar-thumb {
  border-radius: 1rem 1rem 1rem;

  background-color: var(--invisible);
  border: 3px solid var(--softblue);
}

/* GENERAL STYLING */
h1,
h2,
h3 {
  max-height: 100%;
  font-family: var(--thematic);
}

h1 {
  font-weight: 200;
  color: var(--white);
}

h3 {
  font-family: var(--thematic);
  font-weight: 400;
  font-size: 1.5vh;
  line-height: 1em;
}

button,
[type="submit"] {
  background-color: var(--green);
  border-radius: var(--small_radius);
  border-color: var(--softblue);

  color: var(--white);
  font-size: 12px;
  font-family: var(--thematic);
  font-weight: 500;
}

.loadingMessage {
  font-weight: bold;
}

/* EVENT vs. SUGGESTION vs. TASKS */

.suggestion {
  box-sizing: border-box;
  padding: 2px;
  background: url("https://www.transparenttextures.com/patterns/black-twill.png"),
    linear-gradient(
      to bottom right,
      var(--transparent_yellow),
      var(--transparent_orange),
      var(--transparent_bloodorange),
      var(--transparent_green)
    );
  background-repeat: repeat, no-repeat;
  background-position: center;
  background-clip: padding-box, border-box;
  background-blend-mode: difference, normal;
  border: 2px solid var(--transparent_white);

  color: var(--white);
  font-family: var(--thematic);
  font-size: 13px;
  font-weight: 700;
  text-shadow: -1px -1px 0 var(--transparent_navy),
    1px -1px 0 var(--transparent_navy), -1px 1px 0 var(--transparent_navy),
    1px 1px 0 var(--transparent_navy), 0 0 0.25rem var(--transparent_navy);
}

.suggestion p {
  padding: 1px;
  background-color: var(--transparent_white);
  box-shadow: 0 0 0.25rem var(--transparent_white),
    0 0 0.5rem var(--transparent_white), 0 0 0.75rem var(--transparent_white);
  border-radius: var(--small_radius);
}

/* LANDING PAGE */
.guestLanding {
  margin: auto;
  padding: 12.5%;
  width: fit-content;
  max-width: 25em;

  border-radius: var(--large_radius);
  box-shadow: var(--large_shadow);

  background-color: var(--transparent_softblue);
  text-align: center;
}

.guestLanding h2 {
  color: var(--white);
}

.guestLanding p {
  margin: 10% auto;
  width: 85%;

  border-radius: var(--small_radius);
  background-color: var(--transparent_white);
}

.login {
  padding: 5%;
}

.login button,
.login .buttonLabel {
  margin: 1%;
}

/* MAIN APP COMPONENTS */

main.app {
  display: grid;
  box-sizing: border-box;
  position: absolute;
  margin: auto;
  padding: 0.5em;
  height: 100vh;
  width: 90vw;

  grid-template-rows: 10vh 5vh auto;
}

.app h1 {
  position: relative;
  margin: auto;
  grid-row: 1 / span 1;
}

/* MAIN NAVBAR */
.navBar {
  display: block;
  box-sizing: border-box;
  position: absolute;
  bottom: -3px;
  width: 60vw;
  height: 100%;

  grid-row: 2 / span 1;
}

button.navTab {
  display: block;
  position: relative;
  float: left;
  box-sizing: border-box;
  height: 100%;
  min-width: 25%;
  width: auto;
  max-width: 1fr;

  background: var(--paper), var(--orange);
  background-blend-mode: multiply, normal;
  background-position: center;
  background-size: cover, cover;
  border: var(--app_border);
  border-width: 3px 3px 0px 3px;
  border-top-left-radius: 1em 1em;
  border-top-right-radius: 1em 1em;
  border-bottom-left-radius: 0em;
  border-bottom-right-radius: 0em;
}

.navTab.toggled {
  z-index: 2;
}

.navTab.notToggled {
  border: var(--notToggled_border);
  background-color: var(--bloodorange);
  filter: drop-shadow(0.5em 0.75em 0.5em var(--shadow));
}

.navTab img {
  display: block;
  position: absolute;
  top: 25%;
  left: 25%;
  right: 25%;
  margin: auto 0;
  height: 50%;

  clip-path: circle(50% at 50% 50%);

  opacity: 0.5;
}

nav {
  display: block;
  z-index: 1;
  top: 50%;
  bottom: 50%;
  left: 25%;
  right: 25%;
  margin: auto 0;

  font-family: var(--thematic);
  font-weight: 500;
  font-style: normal;
  color: var(--white);
  font-size: minMax(2em, 120%);
}

/* PLANNERR - MAIN CONTAINER */

.planner {
  display: block;
  box-sizing: border-box;
  position: absolute;
  padding: 0.5rem;
  width: 60vw;
  height: 75vh;

  grid-row: 3 / span 1;

  background: var(--paper),
    linear-gradient(to bottom left, floralwhite 0%, var(--softblue));
  background-blend-mode: multiply, normal;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  filter: drop-shadow(0.5em 0.75em 0.5em var(--shadow));

  border: var(--app_border);

  border-top-right-radius: 1em 1em;
  border-bottom-left-radius: 1em 1em;
  border-bottom-right-radius: 1em 1em;
}

/* CALENDAR COMPONENTS */
.calendar {
  display: inline-grid;
  box-sizing: border-box;
  position: relative;
  margin: auto;
  padding: 0px;
  height: 95%;
  width: 100%;

  grid-template-rows: minmax(5%, 10%) minmax(90%, 95%);
  grid-template-columns: 1.5fr 8.5fr;
  row-gap: 2.5%;
  column-gap: 2.5%;
}

.calendar > * {
  filter: drop-shadow(var(--small_shadow));
}

.dateTimeHeader {
  display: inline-block;
  position: absolute;
  box-sizing: border-box;
  margin: 0%;
  height: 100%;
  width: 100%;
  grid-row: 1 / span 1;

  background-clip: border-box;
  background: linear-gradient(to bottom, var(--bloodorange), var(--orange));
  border-top-left-radius: 0.25em;
  border-top-right-radius: 0.25em;

  color: var(--white);
  text-align: center;
  font-weight: 600;
}

.timeLine .dateTimeHeader {
  color: var(--invisible);
  background-color: var(--invisible);
}

/* CALENDAR NAV */
div.day.navBar {
  display: grid;
  box-sizing: border-box;
  position: relative;
  margin: auto;
  width: 100%;
  height: 100%;

  grid-template-columns: minmax(10%, 15%) minmax(70%, 80%) minmax(10%, 15%);
  grid-column: 2 / span all;
  justify-content: space-evenly;
  align-content: center;

  background-color: var(--bloodorange);
  border-radius: var(--small_radius);

  color: var(--white);
  text-align: center;
  line-height: 1em;
}

.day.navBar button {
  margin: auto;
  padding: 3px;
  max-height: 70%;

  background: linear-gradient(to bottom, var(--orange) 1%, var(--bloodorange));
  box-shadow: var(--small_shadow);
  border: 2px double var(--transparent_bloodorange);

  color: ghostwhite;
}

.calViewOptions {
  display: block;
  box-sizing: border-box;
  margin: auto;
  padding: 5%;
  width: 100%;
  height: 100%;

  background-color: var(--softblue);
  border-radius: var(--small_radius);

  text-align: center;
  font-size: 11px;
}

.calViewOptions button.navTab {
  box-sizing: border-box;
  margin: 2.5%;
  padding: 1%;
  height: 50%;
  width: 25%;

  border: 1px solid var(--bloodorange);
  border-radius: var(--small_radius);
  background: none;
  box-shadow: var(--small_shadow);
  filter: none;

  text-align: center;
  font-size: 0.75rem;
  font-weight: bolder;
  color: var(--bloodorange);
}

.calViewOptions .navTab.toggled {
  color: var(--green);
  border-color: var(--green);
}

.calViewOptions .navTab.notToggled {
  color: var(--transparent_navy);
  border-color: var(--transparent_navy);
  background-color: var(--transparent_green);
}

/* TIMELINE */

.timeLine {
  display: inline-grid;
  box-sizing: border-box;
  position: absolute;
  margin: auto;
  padding: 0% 0.5%;
  width: 14.5%;
  max-width: 15%;
  height: 100%;

  grid-template-columns: 1fr 1fr;
  grid-column: 1;
  grid-row: 2 / span 1;
  row-gap: 1px;
  justify-content: space-evenly;
  align-content: space-between;

  border-top: 3px solid #e86f51;
  background-color: var(--orange);

  color: ghostwhite;
  text-align: right;
  font-family: "Courier New", Courier, monospace;
  font-weight: bold;

  border-radius: var(--small_radius);
}

.timeNotch {
  display: inline-block;
  box-sizing: border-box;
  position: absolute;
  min-height: 1fr;
  height: 100%;
  max-height: 100%;
  max-width: 100%;
  width: 100%;

  border-top: 1px solid ghostwhite;
}

.timeNotch .time {
  display: inline-block;
  position: absolute;
  box-sizing: border-box;
  z-index: 5;
  top: -0.5em;
  bottom: 0.5em;
  left: 5%;
  margin: auto auto 5%;
  padding: -1px;
  height: auto;

  font-size: 12px;
  font-weight: bold;
  text-shadow: -2px -2px 0 var(--orange), 2px -2px 0 var(--orange),
    -2px 2px 0 var(--orange), 2px 2px 0 var(--orange);
  /*background-color: var(--transparent_orange);*/
  background-clip: content-box;
  filter: drop-shadow(0rem 0rem 0.25rem var(--orange));
  border-radius: var(--small_radius);
}

.timeNotch .notch {
  display: none;
}

.hidden {
  color: rgba(232, 196, 104, 0);
}

.timeIndicator {
  display: inline-block;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  padding: 2%;
  height: 175%;
  width: 100%;

  border-style: none;
  background-color: var(--transparent_softblue);
  filter: drop-shadow(0.125em 0.125em 0.125em var(--shadow));
  border-radius: var(--small_radius);

  overflow: visible;

  color: var(--orange);
  font-family: var(--thematic);
  font-weight: bolder;
  text-shadow: 0rem 0rem 0.175rem var(--navy);
}

.timePointer {
  display: inline-block;
  position: absolute;
  z-index: 10;
  bottom: -50%;
  left: 70%;
  width: 50%;
  height: 200%;
  background-color: var(--bloodorange);
  clip-path: polygon(0 1%, 100% 50%, 0% 100%, 12% 51%);
}

.currentTime {
  position: absolute;
  display: inline-block;
  left: 10%;
  bottom: 50%;

  line-height: 0em;
  font-weight: bolder;
  font-size: 12px;
  text-shadow: -1px -1px 0 var(--transparent_green),
    1px -1px 0 var(--transparent_green), -1px 1px 0 var(--transparent_green),
    1px 1px 0 var(--transparent_green);
}

/* DAY + TIMEBLOCKS */
div.dayContainer {
  display: grid;
  box-sizing: border-box;
  position: absolute;
  margin: 0px;
  min-height: 100%;
  width: 100%;

  grid-row: 2 / span 1;
  grid-column: 2 / span all;
  grid-template-columns: 100%;
  column-gap: 1%;
  justify-content: space-between;
}

div.day {
  display: grid;
  box-sizing: border-box;
  position: absolute;
  float: left;
  margin: 0px;
  padding: 0% 1%;
  height: 100%;
  max-height: 100%;
  width: 100%;
  max-width: 100%;

  grid-column: auto / span 1;
  row-gap: 1px;
  grid-template-columns: 1fr;
  justify-content: space-evenly;
  align-content: space-between;

  background-color: var(--white);
  border-top: 3px solid var(--bloodorange);

  border-radius: var(--small_radius);
}

div.day p.noResults {
  margin: 50% auto;
  max-width: 80%;
  text-align: center;
}

.timeBlock {
  display: inline-block;
  position: absolute;
  box-sizing: border-box;
  margin: 0px 5%;
  padding: 0.125em;
  min-height: 100%;
  max-height: 100%;
  max-width: 90%;
  height: 100%;
  width: 90%;

  border-radius: var(--small_radius);
  box-shadow: var(--small_shadow);

  color: ghostwhite;

  overflow: hidden;
}

.timeBlock p {
  position: relative;
  margin: auto;
  font-size: 1.0625em;
  vertical-align: middle;
}

.event {
  border-top: 2px solid var(--green);
  background: linear-gradient(
    to bottom,
    var(--softblue),
    var(--transparent_green)
  );
}

.taskBlock {
  border-top: 2px solid var(--bloodorange);
  background: linear-gradient(
    to bottom,
    var(--orange),
    var(--transparent_bloodorange)
  );
}

div.timeBlock.suggestion {
  z-index: 5;
}

/* TASKBOARD COMPONENTS */

table.taskBoard {
  display: table;
  box-sizing: border-box;
  position: absolute;
  left: 5%;
  right: 5%;
  margin: auto;
  padding: 1em;
  height: auto;
  max-height: 95%;
  width: 90%;

  border-collapse: separate;
  border-spacing: 0em 1em;
  vertical-align: middle;
  align-content: center;

  border-radius: var(--small_radius);
  background-color: var(--transparent_white);
  box-shadow: var(--large_shadow);
}

thead {
  display: table-header-group;
  margin: auto;
  width: 100%;
}

thead caption {
  display: table-caption;
  margin: auto;
  width: fit-content;
  text-align: center;
}

.taskBoard tbody {
  max-height: 20%;

  overflow-y: auto;
}

.taskBoard tr {
  display: table-row;
  box-sizing: border-box;
  filter: drop-shadow(0.25em 0.25em 0.25em var(--shadow));
}

.task {
  display: table-row;
  border-radius: 1em 1em 1em;

  background: #e8c468;
  background-clip: border-box;

  color: #274754;
}

td {
  display: table-cell;
  margin: auto;
  padding: 0.5em;
}
td:first-child {
  border-top-left-radius: 1em;
  border-bottom-left-radius: 1em;
}

td:last-child {
  border-top-right-radius: 1em;
  border-bottom-right-radius: 1em;
}

.task button {
  float: right;
}

.deleteButton {
  color: red;
}

/* Account Panel */

.accountPanel {
  display: block;
  position: absolute;
  margin: 5%;
  padding: 1em;
  width: 80%;
  text-align: center;
  background-color: var(--transparent_white);
  box-shadow: var(--large_shadow);
  border-radius: var(--small_radius);
}

form.userSettingsForm {
  display: block;
  margin: auto;
  max-width: 70%;
}

/* DETAILS BOARD */

.detailsBoard {
  display: block;
  box-sizing: border-box;
  position: absolute;
  z-index: 15;
  margin: auto 5vw auto 62.5vw;
  padding: 0.5%;
  width: 100%;
  max-width: 30vw;
  min-height: fit-content;
  height: 100%;
  max-height: 75vh;

  grid-row: 3 / span 1;

  background-color: var(--white);
  border-top: 5px solid var(--orange);
  border-bottom: 5px solid var(--orange);
  border-left: 2em solid var(--yellow);
  box-shadow: var(--large_shadow);
  border-radius: var(--large_radius);

  transition: right 500ms cubic-bezier(0.17, 0.04, 0.03, 0.94),
    height 400ms linear, width 1000ms cubic-bezier(0.17, 0.04, 0.03, 0.94),
    min-width 1000ms cubic-bezier(0.17, 0.04, 0.03, 0.94),
    max-width 1000ms cubic-bezier(0.17, 0.04, 0.03, 0.94),
    margin-top 800ms linear 0.25s;
}

.toggleButton {
  position: relative;
  float: left;
  top: 40%;
  bottom: 40%;
  left: -2.5em;
  padding: 2.5%;
  height: 5em;
  width: 1.75em;

  background: url("./assets/toggleLeft.png"), var(--transparent_bloodorange);
  border: none;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-blend-mode: color;
}

button.exitButton {
  float: right;

  color: var(--bloodorange);
  border-color: var(--orange);
  background-color: var(--transparent_softblue);
}

.detailsBoard .content {
  margin-top: 0;
  min-width: 20em;
  max-height: 80%;
  overflow-y: hidden;
}

.detailsBoard ::-webkit-scrollbar {
  width: 8px;

  background-color: var(--invisible);
}

.detailsBoard ::-webkit-scrollbar-track {
}

.detailsBoard ::-webkit-scrollbar-thumb {
  border-radius: 1rem 1rem 1rem;

  background-color: var(--transparent_orange);
  border: 2px solid var(--softblue);
}

/* TASK DETAILS */
.taskInfo {
}

/* TASK ADD FORM */
.taskAddForm {
  display: grid;
  box-sizing: border-box;
  margin: auto;
  max-height: 80%;
  max-width: 80%;

  grid-template-columns: auto auto;
  row-gap: 0.75em;
  column-gap: 0.5em;
  justify-content: space-around;
  align-content: space-around;

  font-size: 1.25em;
  text-align: right;
}

.taskAddForm .topper,
.taskAddForm h3,
.taskAddForm h4,
.taskAddForm #submitTask {
  display: block;
  margin: auto;
  grid-column: 1 / span 2;
}

.taskAddForm label {
  grid-column: 1;
}

.taskAddForm input {
  grid-column: 2;
}

.calibrator {
  display: block;
  box-sizing: border-box;
  position: absolute;
  height: 97.5%;
  max-height: 97.55%;

  text-align: center;

  overflow-y: hidden;
}

.calibrator p {
  margin: 0% auto 5% auto;
}

.calibrator .detailsHeader {
  height: 20%;
  width: 100%;
}

.calibrator .detailsBody {
  position: absolute;
  box-sizing: border-box;
  max-height: 70%;
  overflow-y: scroll;
}

table.suggestionsTable {
  display: table;
  position: relative;
  margin: auto;

  overflow-y: scroll;
  border-collapse: separate;
  border-spacing: 0px;
}

table.suggestionsTable tr.suggestion {
  display: table-row;
  margin-top: 3px;
}

.app p {
  display: block;
  width: 80%;
}

@media screen and (orientation: portrait) and (max-width: 600px) {
  html {
    padding: 0vw;
    height: 100vh;
    background-size: cover;
    background-position: bottom right;
  }

  main.app {
    top: 0vh;
    bottom: 0vh;
    margin: 0px;
    height: 100vh;
  }

  div.planner {
    bottom: 0vh;
    margin: 0;
    padding: 1% 0px;
    height: 82.5vh;

    border-radius: 0em;
    border-left: none;
    border-right: none;
  }

  main.app,
  div.planner,
  div.navBar {
    position: absolute;
    left: 0vw;
    width: 100vw;
  }

  .calendar {
    margin: 0px;
    padding: 0%;

    grid-template-rows: 7.5% 92.5%;
    grid-template-columns: 17.5vw 75vw;
    column-gap: 2.5vw;
    justify-content: flex-start;
  }

  .timeLine {
    left: 0vw;
    width: 17.5vw;
    max-width: 17.5vw;
    border-top-left-radius: 0em;
    border-bottom-left-radius: 0em;
  }

  .time {
    left: 0vw;
  }

  .detailsBoard {
    min-width: 80vw;
    height: 85vh;
    max-height: 95vh;

    /*transition: right 0.5s cubic-bezier(0.17, 0.04, 0.03, 0.94);*/
  }
}
