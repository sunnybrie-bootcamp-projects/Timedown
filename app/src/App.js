import "./App.css";

import * as React from "react";

import * as ApiClient from "./ApiClient";
import gcal from "./api/ApiCalendar";
import Planner from "./components/Planner.js";

const App = () => {
  const [isAuthenticated, setIsAuthenticated] = React.useState(gcal.sign);
  const [user, setUser] = React.useState({});
  const [loggedIn, setLoggedIn] = React.useState(false);
  const [loginMessage, setLoginMessage] = React.useState("Get started!");

  async function getUserInfo(account) {
    switch (account) {
      case "google":
        if (isAuthenticated) {
          let googleUserInfo = await gcal.getBasicUserProfile();
          setUser({ ...user, google: googleUserInfo });
          //console.debug(googleUserInfo);
        }
        break;
      case "timedown":
        if (isAuthenticated && user.google) {
          if (user.new) {
            let register = await ApiClient.addUser(
              user.google.getEmail(),
              user.google.getGivenName(),
            );
            setLoginMessage(register.message);
            setUser({ ...user, new: false, timedown: register.account });
          } else {
            let timedownUserInfo = await ApiClient.getUser(
              user.google.getEmail(),
            );
            if (timedownUserInfo.error) {
              setLoginMessage(timedownUserInfo.error);
              gcal.handleSignoutClick();
              setUser({});
            } else {
              setUser({ google: user.google, timedown: timedownUserInfo });
              setLoginMessage("Loading account information...");
            }
          }
        }

        break;
      default:
        break;
    }
  }

  //Authenticate client
  React.useEffect(() => {
    gcal.onLoad(() => {
      setIsAuthenticated(gcal.gapi.auth2.getAuthInstance().isSignedIn.get());
      gcal.listenSign((sign) => setIsAuthenticated(sign));
    });
  }, []);

  //load google user info
  React.useEffect(() => {
    getUserInfo("google");
  }, [isAuthenticated]);

  //fetch timedown account
  React.useEffect(() => {
    getUserInfo("timedown");
  }, [user.google]);

  //logged in!
  React.useEffect(() => {
    if (user.timedown) {
      setLoggedIn(true);
    } else {
      setLoggedIn(false);
    }
  }, [user.timedown]);

  if (loggedIn) {
    return (
      <UserDashboard
        {...{
          isAuthenticated,
          gcal,
          setUser,
          user,
          setLoggedIn,
        }}
      />
    );
  }

  return (
    <main className="App">
      <h2>Welcome to Timedown!</h2>
      <p>
        Timedown is an app designed to make it easier to manage your time. Log
        in or register, add a task to your taskboard, and get suggestions for
        times to work on it so you can finish it by its duedate!
      </p>
      <p>{loginMessage}</p>
      <Login {...{ loggedIn, gcal, setUser, user }} />
    </main>
  );
};

function UserDashboard({
  isAuthenticated,
  loggedIn,
  setLoggedIn,
  gcal,
  user,
  setUser,
}) {
  return (
    <main className="app">
      <h1>Timedown</h1>
      <Planner
        {...{ isAuthenticated, loggedIn, setLoggedIn, gcal, user, setUser }}
      />
    </main>
  );
}

const Login = ({ loggedIn, gcal, setUser, user }) => {
  return (
    <div className="login">
      Returning Users:{" "}
      <button onClick={gcal.handleAuthClick}>Log in with Google</button>
      Guests:{" "}
      <button
        onClick={() => {
          setUser({ ...user, new: true });
          gcal.handleAuthClick();
        }}
      >
        Register with Google
      </button>
    </div>
  );
};

export default App;
