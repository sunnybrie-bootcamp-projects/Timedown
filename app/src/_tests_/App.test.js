import React from "react";

import TestRenderer from "react-test-renderer";

import App from "../App";

jest.mock("../ApiClient");

it("Renders greeting messages", () => {
  const testRenderer = TestRenderer.create(<App />);
  const testInstance = testRenderer.root;
  expect(testInstance.findAllByType("p").length).toBe(2);
});

it("Renders login buttons", () => {
  const testRenderer = TestRenderer.create(<App />);
  const testInstance = testRenderer.root;
  expect(testInstance.findAllByType("button").length).toBe(2);
});

it("Renders consistently", () => {
  const testRenderer = TestRenderer.create(<App />).toJSON();
  expect(testRenderer).toMatchSnapshot();
});

it("Renders title", () => {
  const testRenderer = TestRenderer.create(<App />);
  const testInstance = testRenderer.root;
  expect(testInstance.findAllByType("h2").length).toBe(1);
});
