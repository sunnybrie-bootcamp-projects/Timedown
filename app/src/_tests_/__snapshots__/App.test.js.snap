// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders consistently 1`] = `
<main
  className="App"
>
  <div
    className="guestLanding"
  >
    <h2>
      Welcome to Timedown!
    </h2>
    <p>
      Timedown is an app designed to make it easier to manage your time. Log in or register, add a task to your taskboard, and get suggestions for times to work on it so you can finish it by its duedate!
    </p>
    <p
      className="loadingMessage"
    >
      Get started!
    </p>
    <div
      className="login"
    >
      <span
        className="buttonLabel"
      >
         Members:
      </span>
      <button
        onClick={[Function]}
      >
        Log in with Google
      </button>
      <br />
      <span
        className="buttonLabel"
      >
         Guests: 
      </span>
      <button
        onClick={[Function]}
      >
        Register with Google
      </button>
    </div>
  </div>
</main>
`;
